An error occurred while reranking papers

This project will be working normally until February 28th. But after the recent update, it cannot run normally and the following error occurs. It seems that there was an error in the reranking papers stage, with 'SentenceTransformer' issues.

2026-03-02 09:39:29 | INFO | zotero_arxiv_daily.executor:run:78 - Reranking papers...

Error executing job with overrides: []

Traceback (most recent call last):

File "/home/runner/work/zotero-arxiv-daily/zotero-arxiv-daily/src/zotero_arxiv_daily/main.py", line 32, in main

executor.run()

~~~~~~~~~~~~^^

File "/home/runner/work/zotero-arxiv-daily/zotero-arxiv-daily/src/zotero_arxiv_daily/executor.py", line 79, in run

reranked_papers = self.reranker.rerank(all_papers, corpus)

File "/home/runner/work/zotero-arxiv-daily/zotero-arxiv-daily/src/zotero_arxiv_daily/reranker/base.py", line 14, in rerank

sim = self.get_similarity_score([c.abstract for c in candidates], [c.abstract for c in corpus])

File "/home/runner/work/zotero-arxiv-daily/zotero-arxiv-daily/src/zotero_arxiv_daily/reranker/local.py", line 22, in get_similarity_score

encoder = SentenceTransformer(self.config.reranker.local.model, trust_remote_code=True)

^^^^^^^^^^^^^^^^^^^

UnboundLocalError: cannot access local variable 'SentenceTransformer' where it is not associated with a value



Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

Error: Process completed with exit code 1.

--- a/src/zotero_arxiv_daily/reranker/local.py
+++ b/src/zotero_arxiv_daily/reranker/local.py
@@ -5,10 +5,10 @@ import numpy as np
 @register_reranker("local")
 class LocalReranker(BaseReranker):
     def get_similarity_score(self, s1: list[str], s2: list[str]) -> np.ndarray:
+        from sentence_transformers import SentenceTransformer
         if not self.config.executor.debug:
             from transformers.utils import logging as transformers_logging
             from huggingface_hub.utils import logging as hf_logging
-            from sentence_transformers import SentenceTransformer
     
             transformers_logging.set_verbosity_error()
             hf_logging.set_verbosity_error()

